File: 2021/src/test/java/nel/marco/Day13Test.java
package nel.marco;

import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.*;

class Day13Test {

    Day13 day13 = new Day13();

    public List<String> readInput(boolean readExample) throws IOException {

        String path = "src\\main\\resources\\day13";
        if (readExample) {
            path = "src\\main\\resources\\day13Example";
        }

        List<String> stringList = Files.readAllLines(Path.of(path));

        return new ArrayList<>(stringList);

    }

    @Test
    void part1() throws IOException {
        assertEquals(-1, day13.part1(readInput(true)));
        assertEquals(-1, day13.part1(readInput(false)));
    }

    @Test
    void part2() throws IOException {
        assertEquals(-1, day13.part2(readInput(true)), "Part 2 Example message is wrong");//6231
        assertEquals(-1, day13.part2(readInput(false)),"Part 2 actual is wrong");
    }

}

File: 2024/src/test/nel/marco/Day13Test.kt
package nel.marco

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotEquals
import org.junit.jupiter.api.Disabled
import org.junit.jupiter.api.Test

class Day13Test {

    val day = Day13(useExample = false, useMac = false)
    val example = Day13(useExample = true, useMac = false)

    @Test
    fun answerOne() {
        assertEquals("480", example.answerOne())
        assertEquals("40069", day.answerOne())
    }


    @Test
    fun answerTwo() {
        assertEquals("875318608908", example.answerTwo())
        assertEquals("71493195288102", day.answerTwo())
    }


    @Test
    @Disabled
    fun avgTime() {
        MarcoUtil.avgTime("part1") {
            assertEquals("40069", Day13(useExample = false, useMac = false).answerOne())
        }
        MarcoUtil.avgTime("part2") {
            assertEquals("71493195288102", Day13(useExample = false, useMac = false).answerTwo())
        }
    }
}
